### ISSUE TITLE
Screen reader does not announce role of button in CommandBar 

### ISSUE THREAD

**Issue by rachelgshaffer** [microsoft/fluentui#18855]
### Environment Information

- **Package version(s)**: 8.22.0
- **Browser and OS versions**: Chrome 91.0.4472.106, JAWS 2021

### Describe the issue:

When using a screen reader, the role is not announced for buttons in a CommandBar component

### Please provide a reproduction of the issue in a codepen:

This is present in the Fluent samples here: https://fabricweb.z5.web.core.windows.net/oufr/8.22.1/#/examples/commandbar

#### Actual behavior:

Announced like "New 1 of 7", "Upload 2 of 7"

#### Expected behavior:

"New menu item 1 of 7", "Upload menu item 2 of 7"

### Documentation describing expected behavior

MAS reference 04.01.02: Name, Role, Value
Ensure that Assistive Technologies (AT) can gather information about, activate (or set) and keep up to date on the status of user interface controls in the content. For any product, service, tool or content, functionality and navigation of the content and user interface (UI) components must be programmatically exposed and operable.

**Comment by ling1726** [comment]
@rachelgshaffer can you please include some information on the screen reader(s) that you've tested where this issue occurs ?

**Comment by ling1726** [comment]
@smhigley I unfortunately don't have JAWs but the issue will repro with NVDA. However in NVDA this is by by design according to nvaccess/nvda#8654

I couldn't find a similar argument in JAWs issues, hoping you can shed some light on JAWs here.

**Comment by rachelgshaffer** [comment]
Apologies for not including that information. I tested with Chrome + JAW. The accessibility bug opened against my team had the comment "This issue also exists with Firefox + NVDA & Edge(Chromium) + Narrator."

**Comment by ling1726** [comment]
NVDA seems to consider this behaviour as `by design` according to the issue I linked: nvaccess/nvda#8654

However, I can't find any similar statements for other screen readers, will need to defer to @smhigley's expertise on this one I'm afraid
